python/ray/serve/_private/utils.py

```diff
@@ -672,3 +673,19 @@ def call_function_from_import_path(import_path: str) -> Any:
         return callback_func()
     except Exception as e:
         raise RuntimeError(f"The function {import_path} raised an exception: {e}")
+
+
+def get_head_node_id() -> str:
+    """Get the head node id.
+
+    Iterate through all nodes in the ray cluster and return the node id of the first
```

Comment:
curious: doesn't ray guarantee we only have one alive head node?

Reply:
Oo I don't know if we will ever have the case of having multiple alive head nodes, but this is just trying to be accurate if that happens, the first one will be returned 🙃